core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +826,674 @@ private void thereAndBack(byte[] bytes) {
+          .jsonValueAny(
+              SqlFunctions.PathContext
+                  .withReturned(SqlFunctions.PathMode.LAX, Collections.emptyList()),
+              SqlJsonValueEmptyOrErrorBehavior.ERROR,
+              null,
+              SqlJsonValueEmptyOrErrorBehavior.NULL,
+              null);
+      fail("expect exception, but not");
+    } catch (Exception e) {
+      assertThat(e.getMessage(), is("empty json value"));
+    }
+    assertThat(
+        SqlFunctions
+            .jsonValueAny(
+                SqlFunctions.PathContext
+                    .withStrictException(new Exception("test message")),
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null,
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null),
+        nullValue()
+    );
+    assertThat(
+        SqlFunctions
+            .jsonValueAny(
+                SqlFunctions.PathContext
+                    .withStrictException(new Exception("test message")),
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null,
+                SqlJsonValueEmptyOrErrorBehavior.DEFAULT,
+                "empty"),
+        is("empty")
+    );
+    try {
+      SqlFunctions
+          .jsonValueAny(
+              SqlFunctions.PathContext
+                  .withStrictException(new Exception("test message")),
+              SqlJsonValueEmptyOrErrorBehavior.NULL,
+              null,
+              SqlJsonValueEmptyOrErrorBehavior.ERROR,
+              null);
+      fail("expect exception, but not");
+    } catch (Exception e) {
+      assertThat(e.getMessage(), is("java.lang.Exception: test message"));
+    }
+    assertThat(
+        SqlFunctions
+            .jsonValueAny(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.emptyList()),
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null,
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null),
+        nullValue()
+    );
+    assertThat(
+        SqlFunctions
+            .jsonValueAny(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.emptyList()),
+                SqlJsonValueEmptyOrErrorBehavior.NULL,
+                null,
+                SqlJsonValueEmptyOrErrorBehavior.DEFAULT,
+                "empty"),
+        is("empty")
+    );
+    try {
+      SqlFunctions
+          .jsonValueAny(
+              SqlFunctions.PathContext
+                  .withReturned(SqlFunctions.PathMode.STRICT, Collections.emptyList()),
+              SqlJsonValueEmptyOrErrorBehavior.NULL,
+              null,
+              SqlJsonValueEmptyOrErrorBehavior.ERROR,
+              null);
+      fail("expect exception, but not");
+    } catch (Exception e) {
+      assertThat(e.getMessage(), is("java.lang.RuntimeException: not a json value: []"));
+    }
+  }
+
+  @Test public void testJsonQuery() {
```

Comment:
Would you please ensure the tests print human-understandable messages in case of failures?